soh/soh/Enhancements/randomizer/3drando/entrance.cpp

```diff
@@ -971,6 +1001,27 @@ int ShuffleAllEntrances() {
     }
   }
 
+  // Shuffle Bosses
+  if (Settings::ShuffleBossEntrances.IsNot(SHUFFLEBOSSES_OFF)) {
+    if (Settings::ShuffleBossEntrances.Is(SHUFFLEBOSSES_FULL)) {
+      entrancePools[EntranceType::Boss] = GetShuffleableEntrances(EntranceType::ChildBoss);
+      AddElementsToPool(entrancePools[EntranceType::Boss], GetShuffleableEntrances(EntranceType::AdultBoss));
+      // If forest is closed, ensure Ghoma is inside the Deku tree
```

Comment:
could you explain why this is required?

Reply:
closed forest requires that Ghoma is defeated to be opened. So Ghoma needs to be guaranteed to be in the Deku Tree.

There is already an exclusion to remove deku tree from dungeon entrance randomizer when closed forest is on (because of Spirit temple hands).